cfgmgr/teammgr.cpp

```diff
@@ -0,0 +1,485 @@
+                    {
+                        admin_status = fvValue(i);;
+                        SWSS_LOG_INFO("Get admin_status %s",
+                                admin_status.c_str());
+                    }
+                    else if (fvField(i) == "mtu")
+                    {
+                        mtu = fvValue(i);
+                        SWSS_LOG_INFO("Get MTU %s", mtu.c_str());
+                    }
+                }
+
+                if (m_lagList.find(alias) == m_lagList.end())
+                {
+                    addLag(alias, min_links, fallback);
+                    m_lagList.insert(alias);
+                }
+
+                setLagAdminStatus(alias, admin_status);
+                setLagMtu(alias, mtu);
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (m_lagList.find(alias) != m_lagList.end())
+            {
+                removeLag(alias);
+                m_lagList.erase(alias);
+            }
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+void TeamMgr::doLagMemberTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        auto tokens = tokenize(kfvKey(t), '|');
+        auto lag = tokens[0];
+        auto member = tokens[1];
+
+        auto op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            if (!isPortStateOk(member) || !isLagStateOk(lag))
+            {
+                it++;
+                continue;
+            }
+
+            if (m_portList.find(member) == m_portList.end())
+            {
+                addLagMember(lag, member);
+                m_portList.insert(member);
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (m_portList.find(member) != m_portList.end())
+            {
+                removeLagMember(lag, member);
+                m_portList.erase(member);
+            }
+        }
+
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+// When a port gets removed and created again, notification is triggered
+// when state dabatabase gets updated. In this situation, the port needs
+// to be enslaved into the LAG again.
+void TeamMgr::doPortUpdateTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        auto alias = kfvKey(t);
+        auto op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            SWSS_LOG_INFO("Received port %s state update", alias.c_str());
+
+            if (m_portList.find(alias) == m_portList.end())
+            {
+                vector<string> keys;
+                m_cfgLagMemberTable.getKeys(keys);
+
+                for (auto key : keys)
+                {
+                    SWSS_LOG_NOTICE("current key: %s", key.c_str());
```

Comment:
> SWSS_LOG_NOTICE("current key: %s", key.c_str()); [](start = 19, length = 49)

this cannot be notice level

Reply:
debug level.

---
In reply to: [219432425](https://github.com/Azure/sonic-swss/pull/626#discussion_r219432425) [](ancestors = 219432425)